src/broker/broker_monitor.c

```diff
@@ -1943,6 +1985,279 @@ endwin ()
   clear ();
   move (0, 0);
 }
+#else
+static void
+refresh ()
+{
+  fflush (stdout);
+}
+
+static void
+move (int x, int y)
+{
+  currentY = y;
+  tputs (tgoto (cm, x, y), 1, putchar);
+  fflush (stdout);
+}
+
+static void
+clrtobot ()
+{
+  tputs (cd, 1, putchar);
+  fflush (stdout);
+}
+
+static void
+clrtoeol ()
+{
+  tputs (ce, 1, putchar);
+  fflush (stdout);
+}
+
+static void
+endwin ()
+{
+  tcsetattr (0, TCSAFLUSH, &oterm);
+  clear ();
+  move (0, 0);
+
+  if (dl_handle != NULL)
+    {
+      dlclose (dl_handle);
+      dl_handle = NULL;
+    }
+}
+
+static void
+addstr (const char *str)
+{
+  /* line# starts at 0 */
+  if (str == NULL || currentY >= tty_Lines)
+    {
+      return;
+    }
+
+  currentY += num_newlines (str);
+
+  if (currentY >= tty_Lines)
+    {
+      return;
+    }
+
+  fprintf (stdout, "%s", str);
+  fflush (stdout);
+}
+
+static int
+getch ()
+{
+  fd_set readfds;
+  struct timeval tv, *tv_p = &tv;
+  int fd = fileno (stdin);
+  int timeout = get_timeout ();
+  int ret = -1;
+
+  if (timeout < 0)
+    {
+      tv_p = NULL;
+    }
+  else
+    {
+      tv.tv_sec = timeout / 1000;
+      tv.tv_usec = timeout % 1000;
+    }
+
+  FD_ZERO (&readfds);
+  FD_SET (fd, &readfds);
+
+  ret = select (fd + 1, &readfds, (fd_set *) 0, (fd_set *) 0, tv_p);
+
+  if (ret <= 0)
+    {
+      return -1;
+    }
+
+  return fgetc (stdin);
+}
+
+static void
+clear ()
+{
+  currentY = 0;
+  tputs (cl, 1, putchar);
+  fflush (stdout);
```

Comment:
How about adding code "tty_Lines = tgetnum ("li");" for changing line height when user resize terminal ?